tests/phpunit/tests/media.php

```diff
@@ -1155,6 +1155,42 @@ public function test_add_image_size() {
 		$this->assertSame( 600, $sizes['test-size']['height'] );
 	}
 
+	/**
+	 * @ticket 56288
+	 */
+	public function test_add_image_size_additional_mimes() {
+		add_image_size( 'test-size-with-additional-mimes', 200, 600, false, true );
+		add_image_size( 'test-size-without-additional-mimes', 200, 600, false, false );
+
+		$sizes = wp_get_additional_image_sizes();
+
+		remove_image_size( 'test-size-with-additional-mimes' );
+		remove_image_size( 'test-size-without-additional-mimes' );
+
+		$this->assertArrayHasKey( 'test-size-with-additional-mimes', $sizes );
+		$this->assertArrayHasKey( 'test-size-without-additional-mimes', $sizes );
+		$this->assertTrue( $sizes['test-size-with-additional-mimes']['additional_mimes'] );
+		$this->assertFalse( $sizes['test-size-without-additional-mimes']['additional_mimes'] );
+	}
+
+	/**
+	 * @ticket 56288
+	 */
+	public function test__wp_filter_image_sizes_additional_mime_type_support_with_add_image_size() {
+		add_image_size( 'test-size-with-additional-mimes', 200, 600, false, true );
+		add_image_size( 'test-size-without-additional-mimes', 200, 600, false, false );
+
+		$all_sizes      = wp_get_additional_image_sizes();
+		$filtered_sizes = _wp_filter_image_sizes_additional_mime_type_support( $all_sizes, $this->large_id );
+
+		remove_image_size( 'test-size-with-additional-mimes' );
+		remove_image_size( 'test-size-without-additional-mimes' );
+
+		$this->assertArrayHasKey( 'test-size-with-additional-mimes', $filtered_sizes );
+		$this->assertArrayNotHasKey( 'test-size-without-additional-mimes', $filtered_sizes );
```

Comment:
Instead of those specific assertions, I think it would be a bit more precise and comprehensive to build up the full expected array here and then ensure that all expected sizes are included. Something like
```php
$this->assertSame( $expected_size_names, array_keys( $filtered_sizes ) );
```

Reply:
👍🏼 Good idea, updated in 2474368056